hadoop-hdfs-project/hadoop-hdfs/src/test/resources/log4j.properties

```diff
@@ -49,4 +56,24 @@ log4j.appender.DNMETRICSRFA.MaxBackupIndex=1
 log4j.appender.DNMETRICSRFA.MaxFileSize=64MB
 
 # Supress KMS error log
-log4j.logger.com.sun.jersey.server.wadl.generators.WadlGeneratorJAXBGrammarGenerator=OFF
\ No newline at end of file
+log4j.logger.com.sun.jersey.server.wadl.generators.WadlGeneratorJAXBGrammarGenerator=OFF
+
+#
+# hdfs audit logging
+#
+
+# TODO : log4j2 properties to provide example for using Async appender with other appenders
+hdfs.audit.logger=INFO,ASYNCAPPENDER,RFAAUDIT
+hdfs.audit.log.maxfilesize=256MB
+hdfs.audit.log.maxbackupindex=20
+log4j.logger.org.apache.hadoop.hdfs.server.namenode.FSNamesystem.audit=${hdfs.audit.logger}
+log4j.additivity.org.apache.hadoop.hdfs.server.namenode.FSNamesystem.audit=false
+log4j.appender.RFAAUDIT=org.apache.log4j.RollingFileAppender
+log4j.appender.RFAAUDIT.File=${hadoop.log.dir}/hdfs-audit.log
+log4j.appender.RFAAUDIT.layout=org.apache.log4j.PatternLayout
+log4j.appender.RFAAUDIT.layout.ConversionPattern=%m%n
+log4j.appender.RFAAUDIT.MaxFileSize=${hdfs.audit.log.maxfilesize}
+log4j.appender.RFAAUDIT.MaxBackupIndex=${hdfs.audit.log.maxbackupindex}
+log4j.appender.ASYNCAPPENDER=org.apache.log4j.AsyncAppender
```

Comment:
AsyncAppender itself supports logging to file? Or we need to let it wrap another appender?

Reply:
It turns out both log4j1 and log4j2 allows wrapping other appenders within AsyncAppender. However, log4j1 supports this through xml only, whereas log4j2 supports it using both xml as well as properties.

log4j1 [AsyncAppender javadoc](https://logging.apache.org/log4j/1.2/apidocs/org/apache/log4j/AsyncAppender.html):
 
> **Important note**: The AsyncAppender can only be script configured using the [DOMConfigurator](https://logging.apache.org/log4j/1.2/apidocs/org/apache/log4j/xml/DOMConfigurator.html).




Hence I was thinking, instead of creating a new custom appender that eventually wraps other appender with AsyncAppender and make this overall log4j2 migration even more painful, we can wait until we replace log4j.properties with log4j2.properties and then we can configure AsyncAppender to wrap RFA, I have already tested this out locally.

btw this properties file is used for test purpose only.

Reply:
In fact, this is the reason why i have added some TODO comments with this patch as we anyways need to rework on async appender config as part of log4j2 properties. This TODO will help to use appender ref within AsyncAppender in log4j2.properties. Once that work is complete, this TODO also needs to be removed as part of it.

Reply:
We can introduce custom Async appender that wraps RFA:

```
public class AsyncRFAAppender extends AsyncAppender {

  private long maxFileSize = 10*1024*1024;
  private int maxBackupIndex = 1;
  private long nextRollover = 0;
  protected String fileName = null;

  private boolean addRfaToAsync = false;

   public void append(final LoggingEvent event) {
      super.append(event);
   }

  public void setBlocking(final boolean value) {
     addRfaToAsync = true;
     RFA rfa = new RFA(values); // maxFileSize, maxBackupIndex etc
     this.addAppender(rfa);
  }

}

```

But this will be temporary only as log4j2 needs rework anyways.
WDYT? Shall we introduce new custom appender for temporary time duration?